### PR TITLE
feat: implements LocalChannelRegistry

### LINKED ISSUE #60: Implement LocalChannelRegistry
Because local channels are run on the local machine, we need a way to find the local channel. so LocalChannelRegistry is doing this for us.

### PR DESCRIPTION
**Pull request by raspLarry**
close #60

### CODE REVIEW

File: transport/src/main/java/io/el/channel/local/LocalChannelRegistry.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.net.SocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class LocalChannelRegistry {
+
+  private static final ConcurrentMap<LocalAddress, Channel> boundChannels = new ConcurrentHashMap<>();
+
+  public static LocalAddress register(
+      Channel channel, LocalAddress oldLocalAddress, SocketAddress localAddress) {
```
Comment (majecty) [inline transport/src/main/java/io/el/channel/local/LocalChannelRegistry.java:13]: What is the role of the `oldLocalAddress`?

File: transport/src/main/java/io/el/channel/local/LocalChannel.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 public class LocalChannel extends AbstractChannel {
 
-  private enum State {
+  public enum State {
```
Comment (zeroFruit) [inline transport/src/main/java/io/el/channel/local/LocalChannel.java:10]: Why State as public?

Comment (raspLarry) [reply transport/src/main/java/io/el/channel/local/LocalChannel.java]: it's a mistake

File: transport/src/main/java/io/el/channel/local/LocalChannelRegistry.java (hunk 1 above)
Comment (raspLarry) [reply transport/src/main/java/io/el/channel/local/LocalChannelRegistry.java]: it's to check if localAddress is already registered, but as you said I'd better to check outside of this registry class

**Review by majecty — APPROVED** [review]

**Pull request merged**